### PR TITLE
JDK-8292669: IGV: Search not useable when in Overflow Toolbar

### PR DESCRIPTION
**Pull request by tobiasholenstein**
## Overview

The toolbar of the `EditorTopComponent` uses an overflow toolbar when it does not fit in the width of the TopComponent. When the window was not wide enough the `QuickSearch` used to land in the overflow toolbar. Unfortunately, `QuickSearch` stops working in the overflow toolbar.

Now the `QuickSearch` never lands in the overflow toolbar anymore as shown bellow: 
ToolbarOverflow

Since the Netbeans Platform only allows for one occurrence of `QuickSearch`, it is only shown in the `EditorTopComponent` that was selected last. 

## Implementation
We introduce a new `JPanel` named `topPanel` as a parent container. 
topPanel

`topPanel` contains another  `JPanel` `toolbarPanel` and a `Toolbar` `quickSearchToolbar`:
Toolbars

`toolbarPanel` uses the `GridLayout` which gives the original `Toolbar` `toolBar` the maximal available width while still allowing it to overflow. 
On the right we have `quickSearchToolbar`, a `Toolbar` with a single element `quickSearchPresenter`. We give `quickSearchPresenter` a minimum size and use the `BoxLayout` in `quickSearchToolbar` that respects the minimum size of its children.  Therefore the `quickSearchToolbar` never overflows. 
Toolbars_inner

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292669](https://bugs.openjdk.org/browse/JDK-8292669): IGV: Search not useable when in Overflow Toolbar

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [11f02ffe](https://git.openjdk.org/jdk/pull/9940/files/11f02ffebacf7671d0b285ac6bd63ea24346b49b)
 * [Roberto Castañeda Lozano](https://openjdk.org/census#rcastanedalo) (@robcasloz - **Reviewer**)

### Contributors
 * Christian Hagedorn `<chagedorn@openjdk.org>`
 * Tobias Holenstein `<tholenstein@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9940/head:pull/9940` \
`$ git checkout pull/9940`

Update a local copy of the PR: \
`$ git checkout pull/9940` \
`$ git pull https://git.openjdk.org/jdk pull/9940/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9940`

View PR using the GUI difftool: \
`$ git pr show -t 9940`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9940.diff">https://git.openjdk.org/jdk/pull/9940.diff</a>

</details>

### CODE REVIEW

File: src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/EditorTopComponent.java [hunk 1]
```diff
@@ -324,9 +306,24 @@ public void changed(DiagramViewModel source) {
...
-        toolBar.add(quicksearch);
+
+        quickSearchToolbar = new Toolbar();
+        quickSearchToolbar.setLayout(new BoxLayout(quickSearchToolbar, BoxLayout.LINE_AXIS));
+        quickSearchToolbar.setBorder((Border) UIManager.get("Nb.Editor.Toolbar.border")); //NOI18N
+        quickSearchPresenter.setMinimumSize(quickSearchPresenter.getPreferredSize());
+        quickSearchPresenter.setAlignmentX(Component.RIGHT_ALIGNMENT);
+        quickSearchToolbar.add(quickSearchPresenter);
+
+        // Needed to for toolBar to use maximal available width
```
Comment (chhagedorn) [inline src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/EditorTopComponent.java:318]: Typo
```suggestion
        // Needed for toolBar to use maximal available width
```

**Review by chhagedorn — APPROVED** [review]
Looks good, thanks for fixing this! Now I can better use IGV side by side with the code in the IDE.

I've tested it successfully on Linux.

**Comment by tobiasholenstein** [comment]
Updated PR to add the Search when opening a new Tab without first clicking on it

**Review by chhagedorn — APPROVED** [review]

**Review by robcasloz — APPROVED** [review]
The changeset works fine on Windows as well. I just have a style comment: in my opinion, the import ordering should be left as-is (alphabetic order) for consistency with the rest of the IGV project.

**Comment by tobiasholenstein** [comment]
> The changeset works fine on Windows as well. I just have a style comment: in my opinion, the import ordering should be left as-is (alphabetic order) for consistency with the rest of the IGV project.

Thanks for testing on Windows Roberto! 

You are right imports should stay in alphabetic order. I had to configure my IDE to no separate the java.* and javax.* imports. I changed it now

**Review by robcasloz — APPROVED** [review]

**Comment by tobiasholenstein** [comment]
Thanks @chhagedorn and @robcasloz for the reviews!

/integrate

**Pull request closed without merging**